### ISSUE TITLE
Layer-cached omnibus install?

### ISSUE THREAD

**Issue by matschaffer** [portertech/kitchen-docker#163]
More of a question than an issue, but is there some way to avoid installing omnibus every time and instead relying on the docker image cache?

Or should I create/push my own docker image with omnibus pre-installed?

**Comment by coderanger** [comment]
Add the equivalent of https://github.com/poise/poise-boiler/blob/master/lib/poise_boiler/kitchen.rb#L66-L74 to your kitchen.yml, though until the next release there is a bug that break the layer cache unfortunately.

**Comment by tknerr** [comment]
Using the same here:
https://github.com/tknerr/sample-toplevel-cookbook/blob/master/.kitchen.docker.yml

Layer caching works fine with 2.1.0 (which I'm still using due to #160)

**Comment by matschaffer** [comment]
thanks @tknerr - when I try this it gets stuck on this though:

```
Waiting for SSH service on 192.168.59.103:32803, retrying in 3 seconds
```

But I can telnet to that port and get an openssh header just fine. The kithen.yml looks like this:

```
---
driver:
  name: docker

driver_config:
  provision_command:
    - apt-get update && apt-get install net-tools
    - curl -L https://www.opscode.com/chef/install.sh | bash -s -- -v 11.18.0
    - env GEM_HOME=/tmp/verifier/gems GEM_PATH=/tmp/verifier/gems GEM_CACHE=/tmp/verifier/gems/cache /opt/chef/embedded/bin/gem install thor busser busser-serverspec serverspec bundler && chown -R kitchen:kitchen /tmp/verifier
  require_chef_omnibus: false
  use_sudo: false

platforms:
  - name: ubuntu-14.04

suites:
  - name: default
    run_list:
      - recipe[my-recipe]
    attributes:
      my-attributes:
        ...
```

Anything obvious I might be missing?

**Comment by matschaffer** [comment]
Oh, nm. This may be on my end. Debug shows `Too many authentication failures for kitchen` which is something I seem to be getting today after adding one too many ssh keys to my agent.

**Comment by matschaffer** [comment]
Thanks @tknerr 2.1.0 with that config is much nicer. Now to figure out what's taking the chef converge itself so long to start ;)

**Comment by tknerr** [comment]
@matschaffer if you are using chef_zero and have lots of cookbooks to synchronize that might slow down things a little. Maybe you can try with chef_solo and see if that makes a difference?

Apropos caching: this will make the startup super fast, but beyond that you don't have any caching. If you want that, you could:

* use kitchen-vagrant + vagrant docker provider + vagrant-cachier as an alternative
* implement a docker driver for https://github.com/scalefactory/vagrant-multiprovider-snap :-)

**Comment by aaronmcohen** [comment]
It looks like Step 16 (ssh key) which occurs before the Omnibus install will prevent anything afterwards from being cached. Changing the install order may resolve this.

> Step 16 : RUN echo 'ssh-rsa <key> kitchen_docker_key' >> /home/kitchen/.ssh/authorized_keys
> 
>         ---> Running in 95e15be262f6
> 
>         ---> ed95d7dd4df2
> 
>        Removing intermediate container 95e15be262f6

**Comment by coderanger** [comment]
The bug preventing this from working is now fixed in 2.4.0.
